Sources/Tracking/CustomerIO.swift

```diff
@@ -19,70 +18,30 @@ public protocol CustomerIOInstance: AutoMockable {
         // sourcery:Type=AnyEncodable
         // sourcery:TypeCast="AnyEncodable(data)"
         data: RequestBody?,
-        jsonEncoder: JSONEncoder?,
-        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void
+        jsonEncoder: JSONEncoder?
     )
     func clearIdentify()
 }
 
 public extension CustomerIOInstance {
-    /**
-     Identify a customer (aka: Add or update a profile).
-
-     [Learn more](https://customer.io/docs/identifying-people/) about identifying a customer in Customer.io
-
-     Note: You can only identify 1 profile at a time in your SDK. If you call this function multiple times,
-     the previously identified profile will be removed. Only the latest identified customer is persisted.
-
-     - Parameters:
-     - identifier: ID you want to assign to the customer.
-     This value can be an internal ID that your system uses or an email address.
-     [Learn more](https://customer.io/docs/api/#operation/identify)
-     - onComplete: Asynchronous callback with `Result` of identifying a customer.
-     Check result to see if error or success. Callback called on main thread.
-     - jsonEncoder: Provide custom JSONEncoder to have more control over the JSON request body for attributes.
-     */
-    func identify(
+    func identify<RequestBody: Encodable>(
         identifier: String,
-        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void,
+        body: RequestBody,
         jsonEncoder: JSONEncoder? = nil
     ) {
-        identify(identifier: identifier, body: EmptyRequestBody(), onComplete: onComplete, jsonEncoder: jsonEncoder)
+        identify(identifier: identifier, body: body, jsonEncoder: nil)
```

Comment:
why aren't you cascading down the jsonEncoder in some of these?

Reply:
I should be. thanks for that catch.